### ISSUE TITLE
Library evolution support on cocoapods

### ISSUE THREAD

**Issue by aswathr** [danielgindi/Charts#4871]
* [✅] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/jjatie/Charts/blob/master/CONTRIBUTING.md).

## What did you do?

1. Added Charts as a dependency in my framework project with Cocoapods.
2. Switched `BUILD_LIBRARY_FOR_DISTRIBUTION = YES`
3. Build framework

## What did you expect to happen?

Build to succeed

## What happened instead?

Build fails with errors on `SwiftAlgorithms`
<img width="1798" alt="Screenshot 2022-07-26 at 2 01 21 PM" src="https://user-images.githubusercontent.com/10108566/180961392-f8381916-8d1a-420b-b329-2d2a15e14e64.png">

## Charts Environment

**Charts version/Branch/Commit Number: 4.0.3**
**Xcode version: 13.3**
**Swift version: 5**
**Platform(s) running Charts: iOS 15 target (framework)**
**macOS version running Xcode: 12.1**

## Demo Project
[ChartsSampleEvolution.zip](https://github.com/danielgindi/Charts/files/9188096/ChartsSampleEvolution.zip)

## Possible fix
Library evolution can be disabled for SwiftAlgorithms using the post_install hook, which I did.

```
post_install do |installer|
    installer.pods_project.targets.each do |target|
        puts "#{target.name}"
        if target.name == "SwiftAlgorithms"
          target.build_configurations.each do |config|
            config.build_settings['BUILD_LIBRARY_FOR_DISTRIBUTION'] = 'NO'
          end
        end
    end
end
```

but the warning that throws
<img width="1540" alt="Screenshot 2022-07-26 at 2 10 32 PM" src="https://user-images.githubusercontent.com/10108566/180963228-c46e293c-a7b3-4e27-86a1-e41aac9f41ea.png">

can be silenced by changing `import Algorithms` to `@_implementationOnly import Algorithms`, which IMO shouldn't cause any issues as Charts doesn't seem to expose publicly any types of SwiftAlgorithm?

**Comment by matopeto** [comment]
I have same issue, last version that works with BUILD_LIBRARY_FOR_DISTRIBUTION was 4.0.0

**Comment by pykaso** [comment]
Same issue for me.

**Comment by pmairoldi** [comment]
Closed by #4912
